### PR TITLE
Reduce the thread overhead in NodeMonitorUpdater.

### PR DESCRIPTION
**Pull request by deadmoose**
Before, if N nodes connected quickly, this would spawn N threads all
sleeping for 5 seconds before realizing they don't need to do anything.

### CODE REVIEW

File: core/src/main/java/hudson/node_monitors/NodeMonitorUpdater.java [hunk 1]
```diff
@@ -20,7 +22,16 @@
 @Extension
 public class NodeMonitorUpdater extends ComputerListener {
 
-    private final AtomicInteger id = new AtomicInteger();
+    private static final Runnable runnable = new Runnable() {
```
Comment (oleg-nenashev) [inline core/src/main/java/hudson/node_monitors/NodeMonitorUpdater.java]: Minor: I would assign a self-descriptory variable name (in uppercase?)

**Comment by oleg-nenashev** [comment]
LGTM excepting the naming comment above

**Comment by deadmoose** [comment]
I'm open to suggestions for less-terrible names, otherwise...Yo dawg, I heard you like updating monitors so I put a MONITOR_UPDATER in your NodeMonitorUpdater.

**Comment by oleg-nenashev** [comment]
So bad it's good :+1:

**Comment by akshayabd** [comment]
+1

**Comment by oleg-nenashev** [comment]
I'll note it as RFE

**Comment by daniel-beck** [comment]
@oleg-nenashev Do we want this in 1.609.3?

**Comment by oleg-nenashev** [comment]
@daniel-beck 
Because the performance impact may happen only in corner cases. If a regression appears somehow, we may suffer much more. I'd vote for *not* including it to 1.509.3 if there is no agreement on 1.509.4 to become the next cart of Jenkins release train.

**Pull request merged**